applications/StructuralMechanicsApplication/custom_constitutive/truss_constitutive_law.cpp

```diff
@@ -153,7 +153,12 @@ double TrussConstitutiveLaw::CalculateStressElastic(
     double tangent_modulus(0.0);
     CalculateValue(rParameterValues,TANGENT_MODULUS,tangent_modulus);
 
-    const double current_stress = tangent_modulus*current_strain[0];
+    double current_stress = tangent_modulus*current_strain[0];
+    if (this->HasInitialState())
+    {
+      current_stress += this->GetInitialState().GetInitialStressVector()[0];
+    }
```

Comment:
Since the intention of this member function is to compute the elastic stress, I'm not sure whether this is the correct location to add any initial stresses. How was it decided to do the addition here?

Reply:
Basically the initial stress should be added to each stress that is calculated by the constitutive law. Since the elastic stress seems to be the only stress-measure in this truss constitutive law, this is where I thought it should be added.